src/js_sync.hpp

```diff
@@ -168,6 +173,68 @@ class SessionClass : public ClassDefinition<T, WeakSession> {
     static void wait_for_completion(Direction direction, ContextType ctx, ObjectType this_object, Arguments& args);
 };
 
+template <typename T>
+class ClientResetAfterFunctor {
+public:
+    ClientResetAfterFunctor(typename T::Context ctx, typename T::Function after_func)
+        : m_ctx(Context<T>::get_global_context(ctx))
+        , m_func(ctx, after_func)
+    {
+    }
+
+    typename T::Function func() const
+    {
+        return m_func;
+    }
+
+    void operator()(SharedRealm local_realm, SharedRealm remote_realm)
+    {
+        HANDLESCOPE(m_ctx)
+
+        typename T::Value arguments[] = {
+            create_object<T, RealmClass<T>>(m_ctx, new SharedRealm(local_realm)),
+            create_object<T, RealmClass<T>>(m_ctx, new SharedRealm(remote_realm)),
+        };
+        Function<T>::callback(m_ctx, m_func, 2, arguments);
+        local_realm->close();
+        remote_realm->close();
```

Comment:
Once close is called on a sharedRealm, all instances are closed.  I wonder if we need to document that after a reset occurs, the realm needs to be reopened?

Reply:
I'm not sure about the Realm instance lifecycle in js, but the intention of this feature is that the original Realm instance can continue to be used after the reset, without having to reopen it. Can we avoid closing the instance?

Reply:
I need to close these two instances, since they are created on a different thread (the sync client's thread). If I don't close them, I will get `realm::IncorrectThreadException: Realm accessed from incorrect thread`. Notice that the test demonstrate that the Realm doesn't need to be reopened.